Blocks/HTTP/impl/posix_server.h

```diff
@@ -75,16 +75,56 @@ struct HandlerDoesNotExistException : current::net::HTTPException {
   using current::net::HTTPException::HTTPException;
 };
 
+struct ServeStaticFilesFromOptions {
+  std::string url_dirname;
+  std::vector<std::string> index_filenames;
+
+  ServeStaticFilesFromOptions(const std::string& url_dirname,
+                              const std::vector<std::string>& index_filenames)
+      : url_dirname(url_dirname), index_filenames(index_filenames) {}
+
+  ServeStaticFilesFromOptions(const std::string& url_dirname)
+      : ServeStaticFilesFromOptions(url_dirname, {"index.html", "index.htm"}) {}
+
+  ServeStaticFilesFromOptions() : ServeStaticFilesFromOptions(std::string{current::url::GetURLPathSeparator()}) {}
+};
+
 // Helper to serve a static file.
 // TODO(dkorolev): Expose it externally under a better name, and add a comment/example.
 struct StaticFileServer {
   std::string body;
   std::string content_type;
-  explicit StaticFileServer(const std::string& body, const std::string& content_type)
-      : body(body), content_type(content_type) {}
+  std::string filename;
+  bool is_index;
+  explicit StaticFileServer(const std::string& body, const std::string& content_type, const std::string& filename, const bool is_index)
+      : body(body), content_type(content_type), filename(filename), is_index(is_index) {}
   void operator()(Request r) {
     if (r.method == "GET") {
-      r.connection.SendHTTPResponse(body, HTTPResponseCode.OK, content_type);
+      // NOTE(sompylasar): The same `StaticFileServer` is used to serve both directory and file URLs, but we need to distinguish these cases.
+      const std::string& url_original_path = r.url_original.path;
+      if (is_index && url_original_path.back() == current::url::GetURLPathSeparator()) {
+        // NOTE(sompylasar): Serve the index file at the directory URL. Example: `/static/` serves `/static/index.html`.
+        r.connection.SendHTTPResponse(body, HTTPResponseCode.OK, content_type);
+      } else {
+        const bool ends_with_slash_and_filename =
```

Comment:
Just covered this (and much more!) over the phone with @sompylasar. Update coming from his end.